File: src/main/java/org/gangel/graphomance/metrics/TimeGauge.java
package org.gangel.graphomance.metrics;

import com.codahale.metrics.Gauge;
import org.apache.commons.lang3.time.StopWatch;

import java.util.concurrent.TimeUnit;

public class TimeGauge implements Gauge<Double> {

    private StopWatch timer = StopWatch.createStarted();

    private TimeUnit unit = TimeUnit.SECONDS;

    public TimeGauge() {
    }

    public TimeGauge(TimeUnit unit) {
        this.unit = unit;
    }

    @Override
    public Double getValue() {
        return convert(timer.getTime(TimeUnit.MICROSECONDS));
    }

    public Double convert(long d) {
        double m = unit.toMicros(1);
        return d / m;
    }

    public void start() {
        timer = StopWatch.createStarted();
    }

    public void stop() {
        timer.stop();
    }
}

File: src/main/kotlin/org/gangel/graphomance/metrics/TimeGauge.kt
package org.gangel.graphomance.metrics

import com.codahale.metrics.Gauge
import org.apache.commons.lang3.time.StopWatch
import java.util.concurrent.TimeUnit

class TimeGauge : Gauge<Double> {
	private var timer = StopWatch.createStarted()
	private var unit = TimeUnit.SECONDS

	constructor() {}
	constructor(unit: TimeUnit) {
		this.unit = unit
	}

	override fun getValue(): Double {
		return convert(timer.getTime(TimeUnit.MICROSECONDS))
	}

	fun convert(d: Long): Double {
		val m = unit.toMicros(1)
			.toDouble()
		return d / m
	}

	fun start() {
		timer = StopWatch.createStarted()
	}

	fun stop() {
		timer.stop()
	}
}